### DISCUSSION TITLE
Tooltip.Tip is causing compiletime error in Avalonia.Controls.DataGrid DataGridTextColumn

### DISCUSSION THREAD

**Discussion by Bongotainment** [AvaloniaUI/Avalonia#12401] [Q&A]
**Describe the bug**
In Avalonia.Controls.DataGrid version 11.0.1, I encountered a problem with the ToolTips. When I attempt to compile the solution after adding a ToolTip.Tip to a DataGridTextColumn, I receive the following error: 

`MainWindow.axaml(43, 22): [AVLN:0004] Unable to resolve suitable regular or attached property Tip on type Avalonia.Controls:Avalonia.Controls.ToolTip Line 43, position 22.`

I have this axaml code:

            <DataGrid Margin="0" 
                      ItemsSource="{Binding ListItems}" 
                      IsReadOnly="False" 
                      GridLinesVisibility="All"
                      BorderThickness="1" BorderBrush="Gray"
                      HorizontalAlignment="Stretch"
                      VerticalAlignment="Stretch">
                <DataGrid.Columns>
                    <DataGridTextColumn Header="Land" Width="Auto" 
                                        IsReadOnly="True"
                                        ToolTip.Tip="{Binding Path =ListItems.Item.CountryIso}"
                                        Binding="{Binding CountryName }"/>
                    <DataGridTextColumn Header="Regular Expression" 
                                        IsReadOnly="False"
                                        Width="*" 
                                        Binding="{Binding RegularExpression}" />
                    <DataGridCheckBoxColumn Header="Korrekt" Width="Auto" 
                                            IsReadOnly="True"
                                        Binding="{Binding IsValidRegularExpression}" />
                </DataGrid.Columns>
                
                
            </DataGrid>

The ListItems are defined in my MainViewModel:
```
    private ObservableCollection<IslandListItemViewModel> _listItems;
    public ObservableCollection<IslandListItemViewModel> ListItems
    {
        get => _listItems;
        private set => this.RaiseAndSetIfChanged(ref _listItems, value);
    }
```

and my IslandListItemViewModel looks like this:
```
public class IslandListItemViewModel : ViewModelBase 
{
    private string? _regularExpression = null;
    private bool _IsValidRegularExpression = true;
    
    
    
    public String? CountryName { get; set; }
    
    public String? CountryIso { get; set; }
    
    public String? RegularExpression
    {
        get => _regularExpression;
        set
        {
            _regularExpression = value;
            this.RaisePropertyChanged();
            ValidateRegularExpression();
            
        }
    }
    
    public bool IsValidRegularExpression
    {
        get
        {
            return _IsValidRegularExpression;
        }
        set
        {
            _IsValidRegularExpression = value;
            this.RaisePropertyChanged();
        }
    }

    private void ValidateRegularExpression()
    {
        if (RegularExpression == null)
        {
            IsValidRegularExpression = true;
        }
        
        try
        {
            Regex.Match("", RegularExpression);
            //No exception no problem
            IsValidRegularExpression = true;
        }
        catch (ArgumentException)
        {
            // BAD PATTERN: Syntax error
            IsValidRegularExpression = false;
        }
    }
}
```

**To Reproduce**

To reproduce the error, you need a project with Avalonia 11.0.1 and Avalonia.Controls.DataGrid 11.0.1. Additionally, you should include the Avalonia.Themes.Fluent and Avalonia.Themes.Simple libraries, both in version 11.0.1. The documentation seems to suggest that both themes are necessary(?).

After adding the above AXAML code with an appropriate ViewModel, it should reproduce the error code mentioned at the beginning of this bug report.

**Expected behavior**

The DataGrid should behave like the ItemsControl; that means, when using an ItemsControl and adding the Tooltip to the DataTemplate:
1. It uses the IslandListItemViewModel instead of the MainViewModel as its Path.
2. The code compiles successfully and will show a tooltip when hovering over the column.

**Working example in ItemsControl**
```
            <ItemsControl
                          ItemsSource="{Binding ListItems}">
            
                <ItemsControl.ItemTemplate>
                    <DataTemplate DataType="{x:Type vm:IslandListItemViewModel}">
                        <StackPanel Orientation="Horizontal"
                                    HorizontalAlignment="Stretch"
                                    VerticalAlignment="Stretch" Spacing="5">
                            <TextBlock Text="{Binding CountryName}"
                                       Width="100"
                                       VerticalAlignment="Center"
                                       HorizontalAlignment="Center"
                                       ToolTip.Tip="{Binding CountryIso}" />
                            <TextBox Text="{Binding RegularExpression}"
                                     HorizontalAlignment="Stretch"
                                     VerticalAlignment="Stretch"
                                     Width="380"
                                     TextWrapping="WrapWithOverflow"
                                     />
                            <CheckBox IsChecked="{Binding IsValidRegularExpression}"/>
                       </StackPanel>
                    </DataTemplate>
                </ItemsControl.ItemTemplate>
            </ItemsControl>
```

**OS and Version Information**
Desktop: Yes
OS: Mac
Version 11.0.01

**Comment by timunie** [comment]
That's by design. A DataGridColumn is not a "Control" actually, so it cannot have a ToolTip set. However, you can add a Style for `DataGridCell`:

```xml
<DataGrid ItemsSource="{Binding Items}">
    <DataGrid.Styles>
        <Style Selector="DataGridCell.Tip" x:DataType="viewModels:Item">
            <Setter Property="ToolTip.Tip" Value="{Binding SubItems.Count}"></Setter>
        </Style>
    </DataGrid.Styles>
    <DataGrid.Columns>
        <DataGridTextColumn Binding="{Binding Header}" 
                            CellStyleClasses="Tip"/>
    </DataGrid.Columns>
</DataGrid>
```

Note: As this is not a bug, I'll convert it into a Q&A now.
